### PR TITLE
Simplify/refactor #173

### PR DESCRIPTION
**Pull request by dimaqq**

### CODE REVIEW

**Comment by dimaqq** [comment]
@ojii here

**Comment by ojii** [comment]
sorry, I really prefer ensuring correctness at the type level if possible, so I'm going to close this.

File: src/aiodynamo/expressions.py [hunk 1]
```diff
@@ -377,20 +377,24 @@ def encode(self, params: Parameters) -> str:
...
-            if isinstance(other, AndCondition):
-                return AndCondition(self.children.extending(other.children))
-            else:
-                return AndCondition(self.children.appending(other))
-        return AndCondition(MinLen2AppendOnlyList.create(self, other))
+        if isinstance(self, AndCondition) and isinstance(other, AndCondition):
+            return AndCondition(self.children + other.children)
+        elif isinstance(self, AndCondition):
+            return AndCondition(self.children + (other,))
+        elif isinstance(other, AndCondition):
```
Comment (dimaqq) [inline src/aiodynamo/expressions.py:384]: New and improved: #173 missed this case:
`A and (B and C)` 
That's when the user manually adds parens in non-intuitive way.

Comment (ojii) [reply src/aiodynamo/expressions.py:384]: nice, I'll add that to the other PR

**Comment by dimaqq** [comment]
No problem. Consider including the missed logical branch from the comment above.

**Comment by dimaqq** [comment]
> sorry, I really prefer ensuring correctness at the type level if possible, so I'm going to close this.

I was going to suggest a type guard, but that would require py 3.10 ¯\_(ツ)_/¯
https://devguide.python.org/versions/

**Comment by dimaqq** [comment]
Just a thought: another option would be to override AndCondition and OrCondition constructors to ensure the contents is of the correct length.

**Pull request closed without merging**